server/src/internalClusterTest/java/org/opensearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -1933,7 +1934,7 @@ public void testSnapshotCanceledOnRemovedShard() throws Exception {
         unblockNode(repo, blockedNode);
 
         logger.info("--> ensuring snapshot is aborted and the aborted shard was marked as failed");
-        SnapshotInfo snapshotInfo = waitForCompletion(repo, snapshot, TimeValue.timeValueSeconds(60));
+        SnapshotInfo snapshotInfo = snapshotFuture.get().getSnapshotInfo();
```

Comment:
How is the new logic different from the current `waitForCompletion()` implementation? It looks to me like that method will wait for "test-snap" to complete before allowing the test to finish and the cleanup logic to start.

Reply:
I think `snapshotFuture` include the logic of `waitForCompletion`. This change can make it simplified.

Reply:
My question is more about how does this change fix the bug as you describe it since the existing code should have waited for the snapshot to complete before executing the test clean up.

Reply:
`startFullSnapshot()` include the logic of `setWaitForCompletion(true)`. it can avoid the complicate logic of `waitForCompletion(repo, snapshot, TimeValue.timeValueSeconds(60))` which may has defect, `SnapshotsInProgress` https://github.com/opensearch-project/OpenSearch/blob/996d33adb22ac6a523962166f3677f8dc6ac9c1f/test/framework/src/main/java/org/opensearch/snapshots/AbstractSnapshotIntegTestCase.java#L233 may exsit in Cluster State that may fail the clean up logic.

Reply:
What is the defect? It looks to me like that code will wait for the specified snapshot not to be in progress before returning. Can you reproduce this error reliably? I just want to make sure that this change actually fixes the problem that was observed.